File: TA5/tool-evaluation-4/HAMR/examples/test_event_data_port_periodic_domains_VM/both_vm/CAmkES_seL4/src/main/component/both_vm/test_event_data_port_periodic_domains/emitter_t_impl_src_process_src_thread.scala
// #Sireum

package both_vm.test_event_data_port_periodic_domains

import org.sireum._
import both_vm._

// This file will not be overwritten so is safe to edit
object emitter_t_impl_src_process_src_thread {

  def initialise(api: emitter_t_impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_write_port(Base_Types.Integer_8_example())
  }

  def timeTriggered(api: emitter_t_impl_Operational_Api): Unit = {
    // example api usage


  }

  def activate(api: emitter_t_impl_Operational_Api): Unit = { }

  def deactivate(api: emitter_t_impl_Operational_Api): Unit = { }

  def finalise(api: emitter_t_impl_Operational_Api): Unit = { }

  def recover(api: emitter_t_impl_Operational_Api): Unit = { }
}

File: TA5/tool-evaluation-4/HAMR/examples/test_event_data_port_periodic_domains_VM/receiver_vm/CAmkES_seL4/src/c/ext-c/emitter_t_impl_src_process_src_thread/emitter_t_impl_src_process_src_thread.c
#include <emitter_t_impl_src_process_src_thread_api.h>
#include <emitter_t_impl_src_process_src_thread.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "emitter_t_impl_src_process_src_thread.c", "", "receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_initialise_", 0);

  // examples of api setter and logging usage

  S8 t0 = receiver_vm_Base_Types_Integer_8_example(SF_LAST);
  api_put_write_port__receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread(SF t0);

  api_logInfo__receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread(SF string("Example logInfo"));

  api_logDebug__receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread(SF string("Example logDebug"));

  api_logError__receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread(SF string("Example logError"));
}

Unit receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "emitter_t_impl_src_process_src_thread.c", "", "receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_finalise_", 0);
}

Unit receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_timeTriggered_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "emitter_t_impl_src_process_src_thread.c", "", "receiver_vm_test_event_data_port_periodic_domains_emitter_t_impl_src_process_src_thread_timeTriggered_", 0);

}
